atomisticparsers/gromacs/parser.py

```diff
@@ -690,6 +689,27 @@ def parse_thermodynamic_data(self):
                     sec_energy.contributions.append(
                         EnergyEntry(kind=self._metainfo_mapping[key], value=val))
 
+        if not flag_system_time_map:
```

Comment:
you do not need  this, simply use if not system_time_map

Reply:
1. The reason I set this flag is that self._system_time_map is changed by popping values in the above thermo loop. I want to know if self._system_time_map was empty at the beginning, not at the end of the popping.

2. I'm not sure if this ever happens for Gromacs, but for Lammps there are cases where the time/step information is completely unavailable, however, the calculation and system lists are the same length. So, I think we want to take the risk in this case and assume that they are corresponding (this is what we were doing before, such that if you don't have this some of the Lammps tests fail).